crates/bevy_asset/src/processor/mod.rs

```diff
@@ -227,89 +225,129 @@ impl AssetProcessor {
         let processor = _processor.clone();
         IoTaskPool::get()
             .spawn(async move {
-                processor.process_assets().await;
-                processor.listen_for_source_change_events().await;
+                let start_time = std::time::Instant::now();
+                debug!("Processing Assets");
+
+                processor.initialize().await.unwrap();
+
+                let (task_event_sender, task_event_receiver) = async_channel::unbounded();
```

Comment:
Sending these around everywhere creates a lot of noise. I wonder if we could store them on the `AssetProcessor` instead? That would also allow us to do a `start_task` method that just takes the path, removing even more boilerplate.

Reply:
That makes sense. I was considering solving that by having a way to explicitly cancel the current processing (which might be useful by itself), but this works after you added the select so let's keep it.